lib/Sema/ConstraintSystem.h

```diff
@@ -1681,6 +1681,10 @@ class ConstraintSystem {
   /// subsequent solution would be worse than the best known solution.
   bool recordFix(Fix fix, ConstraintLocatorBuilder locator);
   
+  /// If an UnresolvedDotExpr, SubscriptMember, etc has been resolved by the
+  /// constraint system, return the decl that it references.
+  ValueDecl *findResolvedMemberRef(ConstraintLocator *locator);
```

Comment:
Should this also go into CSDiag.h, if its only used there?

Reply:
Personally, this one feels like it's reasonable 'public' API to have on ConstraintSystem, even if it's only used by CSDiag. And this header is also only interior to Sema, so moving the declaration to CSDiag.h doesn't restrict its scope that much more. If you still prefer, I can certainly move it, though.